src/utils.hpp

```diff
@@ -127,20 +133,37 @@ template <class Fn> class scope_exit {
 
 template <typename StringType, typename T>
 StringType to_string(T value)
-    requires std::is_integral_v<T>
+    requires std::is_integral_v<T> && (!std::is_same_v<T, bool>)
 {
     // Maximum number of characters required to represent a 64 bit integer as a string
     // 20 bytes for UINT64_MAX or INT64_MIN + null byte
-    static constexpr size_t uint64_max_chars = 21;
+    static constexpr size_t max_chars = 21;
 
-    std::array<char, uint64_max_chars> str{};
+    std::array<char, max_chars> str{};
     auto [ptr, ec] = std::to_chars(str.data(), str.data() + str.size(), value);
     if (ec == std::errc()) {
         return {str.data(), ptr};
     }
     return {};
 }
 
-template <typename StringType> StringType to_string(bool value) { return value ? "true" : "false"; }
+template <typename StringType, typename T>
+StringType to_string(T value)
+    requires std::is_floating_point_v<T>
+{
+    using char_type = typename StringType::value_type;
+    using traits_type = typename StringType::traits_type;
+    using allocator_type = typename StringType::allocator_type;
+    std::basic_ostringstream<char_type, traits_type, allocator_type> ss;
+    ss << value;
+    return std::move(ss).str();
+}
```

Comment:
Use `std::to_chars`. The performance is an order of magnitude better:

```c++
template <typename StringType, typename T>
StringType to_string(T value)
    requires std::is_floating_point_v<T> &&
             std::is_same_v<
                 StringType,
                 std::basic_string<typename StringType::value_type,
                                   typename StringType::traits_type,
                                   typename StringType::allocator_type>>
{
    using char_type = typename StringType::value_type;
    static_assert(std::is_same_v<char_type, char>);

    static constexpr std::size_t buffer_size =
        sizeof(value) == 16 ? 35 : (sizeof(value) == 8 ? 17 : 9);

    char buffer[buffer_size];
    auto result = std::to_chars(buffer, buffer + sizeof(buffer), value);
    return {buffer, result.ptr};
}
```

Reply:
Thanks, I was using `to_chars` before but libc++ only supports `double` from version 14, so I'm working on a version using concepts to switch between each implementation.